### PR TITLE
Test compatibility with CML2.4

### PR DESCRIPTION
**Pull request by tmikuska**
There's no way to make this compatible with virl2_client of version 2.4.0 but this PR proposes changes to make it compatible at least with 2.4.1.  The compatibility with versions 2.2.0 and 2.3.0 is preserved.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,10 @@ clean-test: ## remove test and coverage artifacts
 	rm -fr .tox/
 	rm -f .coverage
 	rm -fr htmlcov/
+	rm -f 5f0d96.yaml
+	rm -f default_inventory.ini
+	rm -f default_inventory.yaml
+	rm -f default_testbed.yaml
```
Comment (tmikuska) [inline Makefile:27]: After I ran `make test` or `python setup.py test`, these files remained in the directory.  Let me know if I should add them to `.gitignore` instead.

File: Makefile [hunk 2]
```diff
@@ -34,7 +38,7 @@ report: coverage
 	open htmlcov/index.html
 
 test: ## run tests quickly with the default Python
-	python setup.py test
+	python -W ignore::DeprecationWarning setup.py test
```
Comment (tmikuska) [inline Makefile:41]: If tests are not executed with `"-W ignore::DeprecationWarning"` warning control argument, the following tests fail:
~~~
======================================================================
FAIL: test_view_plugin (tests.v2.good_plugins.CMLGoodPluginTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/dev/repos/simple/virl2_client/virlutils/tests/v2/good_plugins.py", line 65, in test_view_plugin
    self.assertEqual("TEST VIEWER\n", result.output)
AssertionError: 'TEST VIEWER\n' != '/root/dev/repos/simple/virl2_client/virl2[204 chars]ER\n'
+ /root/dev/repos/simple/virl2_client/virl2_client/models/lab.py:1037: DeprecationWarning: Labs created in older CML releases (schema version 0.0.5 or lower) are deprecated.  Use labs with schema version 0.1.0 instead.
+   warnings.warn(
  TEST VIEWER


======================================================================
FAIL: test_cml_id (tests.v2.id.CMLIdTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/dev/repos/simple/virl2_client/virlutils/tests/v2/id.py", line 14, in test_cml_id
    self.assertEqual("{} (ID: {})\n".format(self.get_test_title(), self.get_test_id()), result.output)
AssertionError: 'Mock Test (ID: 5f0d96)\n' != '/root/dev/repos/simple/virl2_client/virl2[215 chars]6)\n'
+ /root/dev/repos/simple/virl2_client/virl2_client/models/lab.py:1037: DeprecationWarning: Labs created in older CML releases (schema version 0.0.5 or lower) are deprecated.  Use labs with schema version 0.1.0 instead.
+   warnings.warn(
  Mock Test (ID: 5f0d96)


----------------------------------------------------------------------
Ran 140 tests in 3.731s

FAILED (failures=2)
Test failed: <unittest.runner.TextTestResult run=140 errors=0 failures=2>
error: Test failed: <unittest.runner.TextTestResult run=140 errors=0 failures=2>
~~~

Comment (tmikuska) [reply Makefile:41]: If you didn't want to have it there by default, the other option would be to run tests with the same `PYTHONWARNINGS="ignore::DeprecationWarning"` set in the environment:
~~~
PYTHONWARNINGS="ignore::DeprecationWarning" make test
~~~

**Comment by xorrkaz** [comment]
Merged to my joe-dev branch ahead of the next release (waiting on 2.4.1).  Thanks!

**Pull request merged**